File: src/main/java/com/matag/game/adminclient/TestAdminClient.java
package com.matag.game.adminclient;

import java.util.ArrayList;

import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

import com.matag.adminentities.DeckInfo;
import com.matag.adminentities.PlayerInfo;
import com.matag.game.config.ConfigService;
import com.matag.game.security.SecurityToken;

@Profile("test")
@Component
public class TestAdminClient extends AdminClient {
  public static boolean playerJoined = false;

  public TestAdminClient(ConfigService configService) {
    super(configService);
  }

  public DeckInfo getDeckInfo(SecurityToken token) {
    return new DeckInfo(new ArrayList<>());
  }

  public PlayerInfo getPlayerInfo(SecurityToken token) {
    if (!playerJoined) {
      playerJoined = true;
      return new PlayerInfo("Guest-1");

    } else {
      return new PlayerInfo("Guest-2");
    }
  }
}

File: src/main/kotlin/com/matag/game/adminclient/TestAdminClient.kt
package com.matag.game.adminclient

import com.matag.adminentities.DeckInfo
import com.matag.adminentities.PlayerInfo
import com.matag.game.config.ConfigService
import com.matag.game.security.SecurityToken
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Profile("test")
@Component
class TestAdminClient(configService: ConfigService) : AdminClient(configService) {
    override fun getDeckInfo(token: SecurityToken?): DeckInfo {
        return DeckInfo(listOf())
    }

    override fun getPlayerInfo(token: SecurityToken?): PlayerInfo {
        if (!playerJoined) {
            playerJoined = true
            return PlayerInfo("Guest-1")
        } else {
            return PlayerInfo("Guest-2")
        }
    }

    companion object {
        var playerJoined: Boolean = false
    }
}
